CefSharp/Internals/MethodRunnerQueue.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright © 2010-2015 The CefSharp Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CefSharp.Internals
+{
+    public sealed class MethodRunnerQueue
+    {
+        private readonly JavascriptObjectRepository repository;
+        private readonly AutoResetEvent stopped = new AutoResetEvent(false);
+        private readonly BlockingCollection<Task<MethodInvocationResult>> queue = new BlockingCollection<Task<MethodInvocationResult>>();
```

Comment:
Just going over this one last time before `merge`. Is it worth simplifying this and just maintaining a collection of `MethodInvocation` objects, as we're basically executing them in order? That would simplify the code.

The sub tasks didn't seem to be getting a cancellation token, I may have missed something obvious as I've only stepped through this a couple of times.